packages/modal/addon/components/nucleus-modal/dialog.js

```diff
@@ -180,6 +188,52 @@ class Dialog extends Component {
     }
   }
 
+  /**
+  * createFocus
+  *
+  * @method createFocus
+  * @private
+  *
+  */
+  _createFocus() {
+    let modalDialog = this.element;
+    let focusEl = modalDialog && modalDialog.querySelector("[autofocus]");
+    if(focusEl) {
+      focusEl.focus();
+    }
+    else {
+      modalDialog.focus();
+    }
+  }
+
+  /**
+  * loopFocus
+  *
+  * @method loopFocus
+  * @private
+  * @param {any} event
+  */
+  loopFocus(event) {
```

Comment:
Should we move loopfocus as a utility and accept the parent selector as an argument?

Reply:
Is that refactor necessary? I am not sure if there are any benefits to it. That would mean, we have to move the `scrolled()` as well, right?

Reply:
Modals might not be the only components that need the focus trap. 
1. Full screen viewers
2. Previews

If we move focus trap as a utility now, saves time to extract it later.

scrolled is very specifc to modals. Its used to achieve the sticky header. Lets ignore that for now?

Reply:
1. Let's not pollute the `@freshworks/core` package since it's consumed by all the other components and there's no tree shaking yet. The objective is to keep the core as lean as possible and only add things as and when they are shared among components.
2. Also, there are no other components in DSM (like previews etc) that might require such a focus trap, at least not yet. So let's keep it simple for now and D.R.Y later.